### ISSUE TITLE
Parallel deployment to same IIS instance - race condition changing `C:\Windows\system32\inetsrv\config\applicationHost.config`

### ISSUE THREAD

**Issue by slawomir-brzezinski-at-interxion** [Aqovia/OctopusPuppet#43]
Hi,
We are using the tool most successfully most of the time, but today we have ran into this concurrency issue.
The readme states:
_'The deployment scheduler will calculate the optimum deployment schedule, by parallezing deployments when it can'_

If deployment to IIS is a first-class citizen here, then it seems to me that Octopus puppet should detect when the packages are deployed to the same IIS instance, as otherwise this exception can occur (*note the timing*):

Deployment of *App1*
```
23:33:11   Info     |       [Log] Creating applicationPool App1.Web
23:33:12   Info     |       [Log] AppPool properties
23:33:12   Info     |       [Log] System.Object[]
23:33:12   Info     |       [Log] [AppPool App1.Web] Setting property processModel.loadUserProfile = false
```

Deployment of *App2*
```
23:33:11   Info     |       [Log] ApplicationPool Container is not safe to remove as it contains other applications
23:33:11   Info     |       [Log] Creating applicationPool App2.Web
23:33:12   Error    |       new-item : Filename: \\?\C:\Windows\system32\inetsrv\config\applicationHost.config
23:33:12   Error    |       Error: Cannot commit configuration changes because the file has changed on disk
23:33:12   Error    |       At C:\services\app2.web\Deployment\PowershellModules\WebInstalltion.psm1:243 char:2
23:33:12   Error    |       +     $appPool = New-WebAppPool -Name $appPoolConfig.name
23:33:12   Error    |       +     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
23:33:12   Error    |       + CategoryInfo          : NotSpecified: (:) [New-Item], FileLoadException
23:33:12   Error    |       + FullyQualifiedErrorId : System.IO.FileLoadException,Microsoft.PowerShell
23:33:12   Error    |       .Commands.NewItemCommand
23:33:12   Error    |       Script 'c:\services\app2.web\Deploy.ps1' returned non-zero exit code: 1. Deployment terminated.
```

I understand that whether deployment is thread-safe depends on the implementation (IIS 8.5.9600.16384 on Windows Server 2012 R2, and [`Powershell.Deployment`](https://github.com/sponte/Powershell.Deployment)) here), but it would be good to have at least a setting 'disallowParallelToSameMachine', as the only other alternative for us then is to disable paralellization.

**Comment by slawomir-brzezinski-at-interxion** [comment]
Turns out this was due to [use of `OctopusBypassDeploymentMutex=true`](http://octopus.office.interxion.net/app#/library/variables/LibraryVariableSets-7) to speed up deployment.

We have now removed it on environments, but this is said to greatly slow down deployments of whole environments. Perhaps there is an opportunity to speed it up by implementing the mutex in [`Powershell.Deployment`](https://github.com/sponte/Powershell.Deployment) scripts, so they are thread safe.
